compact/tracking/mpgd_barrel.xml

```diff
@@ -21,70 +21,91 @@
     <constant name="MMMesh_thickness"                       value="18*um"/>
     <constant name="MMGasGap_thickness"                     value="3000*um"/>
     <constant name="MMDriftCuElectrode_thickness"           value="5*um"/>
     <constant name="MMDriftKapton_thickness"                value="250*um"/>
     <constant name="MMDriftCuGround_thickness"              value="0.41*um"/>
-
     <comment> FIXME: No support material is here, so fudge factor used to bring material budget to ~0.5% for barrel </comment>
-    <constant name="MMFudgeInnerMPGDBarrel_thickness"                      value="570*um"/>
-
+    <constant name="MMFudgeInnerMPGDBarrel_thickness"       value="570*um"/>
+    <comment> FIXME: No definite plan for connections/services to the inner sectors yet, so guess value </comment>
+    <constant name="MMInnerService_thickness"               value="0.05*cm"/>
+    <comment> ...Dimensions </comment>
+    <constant name="MMModuleWidth"                          value="46.0*cm"/>
+    <constant name="MMModuleLength"                         value="61.0*cm"/>
+    <constant name="MMFrameWidth"                           value="1.0*cm"/>
+    <constant name="MMOutwardFrameWidth"                    value="5.0*cm"/>
+    <constant name="MMOuterSector_R"                        value="57.7*cm"/>
+    <constant name="MMInnerSector_R"                        value="55.5*cm"/>
+    <comment> ...Positioning </comment>
+    <constant name="MMRadial_offset"                        value="1.0*cm"/>
+    <constant name="MMCentral_gap"                          value="0.2*cm"/>
 
-    <comment>
-      Extra parameters to approximate a cylinder as a set of skinny staves
-      due to ACTS limitations.
-      FIXME: this shouldn't be needed anymore, need to update the cylindrical plugin.
-    </comment>
-    <constant name="MPGDBarrelStave_count"            value="128"/>
-    <constant name="InnerMPGDBarrelStave_width"       value="2*(InnerMPGDBarrelLayer_rmin) * tan(180*degree/MPGDBarrelStave_count)"/>
+    <comment> In principle, stave count should not be independenty specified: it is fixed by above intrinsic parameters </comment>
+    <constant name="MPGDBarrelStave_count"                  value="8"/>
+    <constant name="InnerMPGDBarrelLayer_rmin"              value="InnerMPGDBarrel_rmin + InnerMPGDBarrel_hack"/>
+    <constant name="InnerMPGDBarrelLayer_rmax"              value="MMOuterSector_R + InnerMPGDBarrel_hack + MMRadial_offset / 2 + 1*cm"/>
+    <constant name="rmin2Sensor" value="MMDriftCuGround_thickness + MMDriftKapton_thickness + MMDriftCuElectrode_thickness + MMGasGap_thickness / 2" />
+    <constant name="MMInnerSensor_R" value="MMInnerSector_R + rmin2Sensor" />
+    <constant name="MMOuterSensor_R" value="MMOuterSector_R + rmin2Sensor" />
   </define>
 
   <detectors>
     <detector
       id="TrackerBarrel_2_ID"
       name="InnerMPGDBarrel"
-      type="epic_TrackerBarrel"
+      type="MPGDCylinderBarrelTracker"
```

Comment:
We prepend `epic_` to the plugin names (detector types) to avoid name clashes when Detector 2 reuses our code (and to avoid clashes with detector types defined centally by DD4hep; though not likely to be an overlap here).

Reply:
But then "./src/MPGDCylinderBarrelTracker_geo.cpp" has to be also modified, hasn't it?

Reply:
Yes, you will need to prepend epic_ to MPGDCylinderBarrelTracker in the .cpp file too

Reply:
Done. See commit #7403586 ("https://github.com/eic/epic/commit/7403586").